prow/pjutil/abort.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pjutil
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+
+	prowapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+)
+
+// prowClient a minimalistic prow client required by the aborter
+type prowClient interface {
+	//ReplaceProwJob replaces the prow job with the given name
+	ReplaceProwJob(string, prowapi.ProwJob) (prowapi.ProwJob, error)
+}
+
+// ProwJobResourcesCleanupFn type for a callback function which it is expected to clean up
+// all k8s resources associated with the given prow job. It should do the best effort to
+// remove these resources, but if for any reason there is an error, it should only log a warning
+// message.
+type ProwJobResourcesCleanupFn func(pj prowapi.ProwJob)
```

Comment:
We don't typically include the type (Fn) in the name

https://dave.cheney.net/practical-go/presentations/qcon-china.html#_dont_name_your_variables_for_their_types

Reply:
Fixed